packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -843,12 +843,25 @@ abstract class ResidentRunner {
 }
 
 class OperationResult {
-  OperationResult(this.code, this.message, { this.hint });
+  OperationResult(this.code, this.message, { this.hint, this.hintId });
 
+  /// The result of the operation; a non-zero code indicates a failure.
   final int code;
+
+  /// A user facing message about the results of the operation.
   final String message;
+
+  /// An optional hint about the results of the operation. This is used to provide
+  /// sidecar data about the operation results. For example, this is used when
+  /// a reload is successful but some changed program elements where not run after a
+  /// reassemble.
   final String hint;
```

Comment:
Maybe rename this `hintMessage` to make it clear that `hintId` is the key used by tools, whereas `hintMessage` is the user-facing message displayed to the user?

Reply:
sgtm!